test/built-ins/Promise/prototype/finally/species-symbol.js

```diff
@@ -0,0 +1,20 @@
+// Copyright (C) 2017 V8. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+author: Sathya Gunasekaran
+description: finally calls the SpeciesConstructor
+esid: sec-promise.prototype.finally
+features: [Promise.prototype.finally]
+---*/
+
+
+class MyPromise extends Promise {
+  static get [Symbol.species]() { return Promise; }
+}
+
+var p = Promise
+    .resolve()
+    .finally(() => MyPromise.resolve());
+
+assert.sameValue(true, p instanceof Promise);
+assert.sameValue(false, p instanceof MyPromise);
```

Comment:
This test seems correct. You can also test the way that Promise is used as the default for SpeciesConstructor (by deleting Promise[Symbol.species] and testing that everything works just as in this test).